### PR TITLE
Cache anthropic

### PR DESCRIPTION
**Pull request by aarondr77**
# Description

Add caching for anthropic. We use 2 cache breakpoints.

1. System prompt (static content) is always cached with a breakpoint, so it's never reprocessed after the first request.
2. Conversation history gets a cache breakpoint on the last message that has passed the trim_old_message threshold, enabling incremental caching. Messages are stable after they are more than MESSAGE_HISTORY_TRIM_THRESHOLD old. At this point, the messages are not edited anymore, so they will not invalidate the cache. If we included the messages before the boundary in the cache, then every time we send a new  message, we would invalidate the cache and we would never get a cache hit except for the system prompt.

Check out this PR too which lets us track cache performance: https://github.com/mito-ds/ai-completion-server/pull/15

# Testing

Try it out, an run the ai-completion-server locally so you can observe the cache hit metrics. 

# Documentation

No. 

---

> [!NOTE]
> Adds Anthropic request caching (system prompt + stable history) and switches message trimming to a global threshold, with supporting type updates and tests.
> 
> - **Anthropic provider**:
>   - Add caching utilities: `add_cache_control_to_message` and `get_anthropic_system_prompt_and_messages_with_caching` (applies `cache_control: {type: "ephemeral"}` to system prompt and the stable-history boundary based on `MESSAGE_HISTORY_TRIM_THRESHOLD`).
>   - Use caching-aware messages in `AnthropicClient.request_completions` and `.stream_completions`.
> - **Message history**:
>   - `trim_old_messages` now uses `MESSAGE_HISTORY_TRIM_THRESHOLD` (removed `keep_recent` param) and preserves recent messages; still trims only user content and simplifies mixed-content trimming.
> - **Utils/Types**:
>   - Update Anthropic utils to accept `system` as `str | List[TextBlockParam] | anthropic.Omit` and propagate to server/stream params.
> - **Constants**:
>   - Add `MESSAGE_HISTORY_TRIM_THRESHOLD = 3`.
> - **Tests**:
>   - Update message history tests to threshold-based API and expand scenarios.
>   - Add caching tests for system prompt and history boundary; adjust provider tests and imports.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit bd4f732cc10570deae0044f617a4bdf4625bd909. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: .cursorrules [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Cursor Rules for Mito Project
+
+## Virtual Environment
+- ALWAYS activate the virtual environment before running any Python commands: `cd mito-ai && source venv/bin/activate`
```
Comment (ngafar) [inline .cursorrules:4]: This rule is a great idea. I can't count the number of times Cursor has installed packages to my base env.
